### PR TITLE
Feat #18 Experts page:  list of authors

### LINKED ISSUE #18: Experts Page: List of Authors
https://www.24life.com/experts/

Pull data from the Author index (https://github.com/hlxsites/24life/issues/14).

### PR DESCRIPTION
**Pull request by iamitshri**
Fix #18 

Test URLs:
- Before: https://main--24life--hlxsites.hlx.page/
- After:  https://18-experts-page--list-of-authors--24life--hlxsites.hlx.page/experts

### CODE REVIEW

File: blocks/author-list/author-list.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  const firstAuthors = authors.slice(0, numInitialLodedAuthors);
+  firstAuthors.forEach((author) => {
+    const newBlock = createAuthorCardBlock(author);
+    block.append(newBlock);
+  });
+  // create load more button if there are more authors than shown
+  if (authors.length > numInitialLodedAuthors) {
+    const { loadMoreButton, loadMoreContainer } = createLoadMoreButton();
+    loadMoreButton.addEventListener('click', () => {
+      // get the next 50 authors
```
Comment (synox) [inline blocks/author-list/author-list.js:31]: Comment does not match code. It seems to fetch everything else

File: blocks/author-list/author-list.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  loadMoreContainer.classList.add('author-load-more-container');
+  const loadMoreButton = document.createElement('button');
+  loadMoreContainer.append(loadMoreButton);
+  loadMoreButton.classList.add('author-list-load-more-button');
+  loadMoreButton.textContent = 'Load more';
+  return { loadMoreButton, loadMoreContainer };
+}
+
+function buildAuthorListItem(className, content) {
+  const list = Array.isArray(content) ? content : [content];
```
Comment (synox) [inline blocks/author-list/author-list.js:55]: This function can probably be simplified. Think a div.append([…content]) should do most of this

File: blocks/author-list/author-list.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    const container = document.createElement('p');
+    container.appendChild(socialLink);
+    socialLinksContainer.appendChild(container);
+  }
+  arr.forEach((link) => {
+    const socialLink = document.createElement('a');
+    socialLink.href = link;
+    socialLink.target = '_blank';
+    if (link.includes('facebook')) {
+      socialLink.innerHTML = '<span class="icon icon-facebook"><svg xmlns="http://www.w3.org/2000/svg"><use href="#icons-sprite-facebook"></use></svg></span>';
```
Comment (synox) [inline blocks/author-list/author-list.js:116]: You can omit the svg code. decorateIcons() will do that for you

File: blocks/author-list/author-list.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+function addAuthorLinks(author, authorLinkContainer) {
+  if (!author.links) return;
+  let arr = JSON.parse(author.links);
+  if (arr.length > 0) {
+    arr = arr[0].split(',');
+  }
+  const socialLinksContainer = document.createElement('div');
+  socialLinksContainer.classList.add('social-links');
+  function addSocialLink(socialLink) {
+    const container = document.createElement('p');
```
Comment (synox) [inline blocks/author-list/author-list.js:107]:
```suggestion
    const container = p();
```

File: styles/styles.css [hunk 5]
```diff
@@ -332,3 +332,17 @@ main .section {
...
+  background: #f8f8f8;
+  letter-spacing: 2px;
+  padding-bottom: 13px;
+  padding-top: 50px;
+}
+
+div.experts-page-editorial-staff {
+  margin: 0;
+  padding: 0;
+  margin-top: -5px;
```
Comment (synox) [inline styles/styles.css:346]: It’s an anti-pattern to have negative values. Maybe we can omit it, or remove space somewhere else. I need to check next week

**Review by synox — CHANGES_REQUESTED** [review]
This looks amazing! Just some minor things

File: blocks/author-list/author-list.js (hunk 3 above)
Comment (iamitshri) [reply blocks/author-list/author-list.js]: fixed

File: blocks/author-list/author-list.js (hunk 2 above)
Comment (iamitshri) [reply blocks/author-list/author-list.js]: fixed

**Review by synox — APPROVED** [review]
I pushed some small changes to handle missing images and overflow on the images.

**Comment by iamitshri** [comment]
@synox Thanks for those changes. I will take care of those things in future PRs

File: blocks/author-list/author-list.css [hunk 6]
```diff
@@ -71,9 +75,9 @@ div.author-list-container {
 }
 
 @media (min-width: 1200px) {
-  .section.experts-page-heading>div,
-  .section.experts-page-editorial-staff>div,
-  .section.author-list-container>div {
+  .section.experts-page-heading > div,
+  .section.experts-page-editorial-staff > div,
+  .section.author-list-container > div {
```
Comment (iamitshri) [inline blocks/author-list/author-list.css:82]: @synox Did you fix this using tool?  my stylint did not do this for me :)

File: blocks/author-list/author-list.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import ffetch from '../../scripts/ffetch.js';
 import {
-  readBlockConfig, loadBlocks, createOptimizedPicture, decorateIcons,
+  createOptimizedPicture, decorateIcons, loadBlocks, readBlockConfig,
 } from '../../scripts/lib-franklin.js';
```
Comment (iamitshri) [inline blocks/author-list/author-list.js:4]: curious was this rearranged using the tool? Asking so I can have a proper tool on my end as well :)

Comment (synox) [reply blocks/author-list/author-list.js:4]: eslint does that. I configured intelliJ to auto-run `eslint --fix` on each save.

File: blocks/author-list/author-list.css (hunk 6 above)
Comment (synox) [reply blocks/author-list/author-list.css:82]: This one I did manually.

**Pull request merged**